core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannel.scala

```diff
@@ -41,7 +40,41 @@ class TransactionMarkerChannel(interBrokerListenerName: ListenerName,
   // we need the metadataPartition so we can clean up when Transaction Log partitions emigrate
   case class PendingTxnKey(metadataPartition: Int, producerId: Long)
 
-  private val brokerStateMap: concurrent.Map[Int, DestinationBrokerAndQueuedMarkers] = concurrent.TrieMap.empty[Int, DestinationBrokerAndQueuedMarkers]
+  class BrokerRequestQueue(private var destination: Node) {
+
+    // keep track of the requests per metadata partition so we can easily clear the queue
+    // during partition emigration
+    private val requestsPerMetadataPartition: concurrent.Map[Int, BlockingQueue[TxnMarkerEntry]]
+      = concurrent.TrieMap.empty[Int, BlockingQueue[TxnMarkerEntry]]
+
+    def removeRequestsForPartition(partition: Int): Unit = {
+      requestsPerMetadataPartition.remove(partition)
+    }
+
+    def maybeUpdateNode(node: Node): Unit = {
+      if (!destination.equals(node))
+        destination = node
+    }
+
+    def addRequests(metadataPartition: Int, txnMarkerEntry: TxnMarkerEntry): Unit ={
+      val queue = requestsPerMetadataPartition.getOrElseUpdate(metadataPartition, new LinkedBlockingQueue[TxnMarkerEntry]())
+      queue.add(txnMarkerEntry)
+    }
+
+    def eachMetadataPartition[B](f:(Int, BlockingQueue[TxnMarkerEntry]) => B): mutable.Iterable[B] =
+      requestsPerMetadataPartition.filter{case(_, queue:BlockingQueue[TxnMarkerEntry]) => !queue.isEmpty}
```

Comment:
It seems like there's a lot of cases where we do `collection.filter{case(foo: Foo, bar: Bar) => ....`. That doesn't follow the usually convention in our Scala code. We typically write it like: `collection.filter { case (foo, bar) => ... }`. That is, we don't usually specify the type and there's a space before and after the brace. Specifying the type in a `case` statement involves a pattern match. What does that mean? If the match fails, you get a runtime error. Example:

```scala
scala> trait Base
defined trait Base

scala> case class Foo(s: String) extends Base
defined class Foo

scala> case class Bar(s: String) extends Base
defined class Bar

scala> Seq[Base](Foo("1"), Foo("2")).map { case (b: Bar) => b }
scala.MatchError: Foo(1) (of class Foo)
  at .$anonfun$res7$1(<console>:17)
  at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
  at scala.collection.immutable.List.foreach(List.scala:389)
  at scala.collection.TraversableLike.map(TraversableLike.scala:234)
  at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
  at scala.collection.immutable.List.map(List.scala:295)
  ... 29 elided
```